gtest_parallel.py

```diff
@@ -204,14 +209,18 @@ def _logname(output_dir, test_binary, test_name, execution_number):
     # Store logs to temporary files if there is no output_dir.
     if output_dir is None:
       (log_handle, log_name) = tempfile.mkstemp(prefix='gtest_parallel_',
-                                                suffix=".log")
-      os.close(log_handle)
-      return log_name
-
-    log_name = '%s-%s-%d.log' % (Task._normalize(os.path.basename(test_binary)),
-                                 Task._normalize(test_name), execution_number)
+                                                suffix='.log')
+    else:
+      suffix = '-%d.log' % execution_number
+      prefix = '%s-%s' % (Task._normalize(os.path.basename(test_binary)),
+                          Task._normalize(test_name))
+      namemax = os.statvfs('.').f_namemax if hasattr(os, 'statvfs') else NAMEMAX_DEFAULT
```

Comment:
Can you have this be: max_prefix_length = be (f_filemax - space_for_random) else default?

Reply:
Cleaned this up